plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1330,4 +1363,26 @@ public long getSnapshotId()
             return this.snapshotId;
         }
     }
+
+    private static final class TableCacheLoader
+            extends CacheLoader<String, ConcurrentHashMap<SchemaTableName, Table>>
+    {
+        @Override
+        public ConcurrentHashMap<SchemaTableName, Table> load(final String unused)
+                throws Exception
+        {
+            return new ConcurrentHashMap<>();
```

Comment:
I think i don't understand what this is doing.

Anyway, i guess this will go away once we move away from unnecessary `Cache` use.

Reply:
The cache builder needs a function as an argument which creates the cache entry if there is a miss for a certain key. Here we simply create a new ConcurrentHashMap if there is no cache for a certain query id, yet.